### DISCUSSION TITLE
Excluding certain routable IP addresses from kube api

### DISCUSSION THREAD

**Discussion by aretecarpe** [siderolabs/talos#9675] [Q&A] [answered]
Hello,

I have a setup in which I have multiple public IP addresses and some private IP addresses on my machine that are routable, I am wondering if it's at all possible to exclude all of those public IP addresses from being signed to the k8s certificates? I would rather not create a nftable rule to only allow my private IP addresses to the kube api.

**Comment by smira** [comment]
Adding or not adding a specific address to the certificate doesn't have anything to do with access to your cluster.

The certificate SANs are checked by the _client_, not the server. So an attacker can simply ignore SAN check.

**Reply by aretecarpe** [reply]
Ah ok, so I guess doing the nftable rule is what I'll do. Do let me know if theres a better way to block out the public IP's from the internal services like talos api and kube api.

**Reply by smira** [reply, marked as answer]
See [Ingress Firewall](https://www.talos.dev/v1.8/talos-guides/network/ingress-firewall/).

Certificate SANs have nothing to do with restricting access.

**Reply by aretecarpe** [reply]
Understood. Thank you.
